### PR TITLE
Make tree equality comparison independent from token's equality comparison

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -186,11 +186,15 @@ def __eq__(self, other):
         if self.token_type == TokenType.STRING:
```
Comment (tobymao) [inline sqlglot/tokens.py:186]: might as well not override __eq__ at all if you're going to check line numbers

Comment (izeigerman) [reply sqlglot/tokens.py:186]: ah, that's true. So we don't really need to normalize any more.

File: sqlglot/parser.py [hunk 2]
```diff
@@ -1054,6 +1069,15 @@ def _parse_tokens(self, parse, expressions):
 
         return this
 
+    def _ensure_non_token(self, value):
```
Comment (tobymao) [inline sqlglot/parser.py:1072]: hmmmmmmmmm

File: tests/test_expressions.py [hunk 3]
```diff
@@ -69,7 +77,7 @@ def test_hash(self):
             {
                 parse_one("select a.b"),
                 parse_one("1+2"),
-                parse_one('"a".b'),
+                parse_one('"a"."b"'),
```
Comment (tobymao) [inline tests/test_expressions.py:80]: this shouldn't have broke, this was an intentional test

Comment (izeigerman) [reply tests/test_expressions.py]: Fixed.

**Review by tobymao — APPROVED** [review]

**Pull request merged**